lib/services/local/mfa_test.go

```diff
@@ -0,0 +1,231 @@
+		},
+		{
+			name: "invalid version",
+			chal: &mfav1.ValidatedMFAChallenge{
+				Kind:     types.KindValidatedMFAChallenge,
+				Version:  "v2",
+				Metadata: &types.Metadata{Name: "test"},
+				Spec: &mfav1.ValidatedMFAChallengeSpec{
+					Payload: &mfav1.SessionIdentifyingPayload{
+						Payload: &mfav1.SessionIdentifyingPayload_SshSessionId{SshSessionId: []byte("id")},
+					},
+					SourceCluster: "src", TargetCluster: "tgt",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing name",
+			chal: &mfav1.ValidatedMFAChallenge{
+				Kind:     types.KindValidatedMFAChallenge,
+				Version:  "v1",
+				Metadata: &types.Metadata{Name: ""},
+				Spec: &mfav1.ValidatedMFAChallengeSpec{
+					Payload: &mfav1.SessionIdentifyingPayload{
+						Payload: &mfav1.SessionIdentifyingPayload_SshSessionId{SshSessionId: []byte("id")},
+					},
+					SourceCluster: "src", TargetCluster: "tgt",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing payload",
+			chal: &mfav1.ValidatedMFAChallenge{
+				Kind:     types.KindValidatedMFAChallenge,
+				Version:  "v1",
+				Metadata: &types.Metadata{Name: "test"},
+				Spec: &mfav1.ValidatedMFAChallengeSpec{
+					Payload:       nil,
+					SourceCluster: "src", TargetCluster: "tgt",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "empty ssh_session_id",
+			chal: &mfav1.ValidatedMFAChallenge{
+				Kind:     types.KindValidatedMFAChallenge,
+				Version:  "v1",
+				Metadata: &types.Metadata{Name: "test"},
+				Spec: &mfav1.ValidatedMFAChallengeSpec{
+					Payload: &mfav1.SessionIdentifyingPayload{
+						Payload: &mfav1.SessionIdentifyingPayload_SshSessionId{SshSessionId: []byte("")},
+					},
+					SourceCluster: "src", TargetCluster: "tgt",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing source_cluster",
+			chal: &mfav1.ValidatedMFAChallenge{
+				Kind:     types.KindValidatedMFAChallenge,
+				Version:  "v1",
+				Metadata: &types.Metadata{Name: "test"},
+				Spec: &mfav1.ValidatedMFAChallengeSpec{
+					Payload: &mfav1.SessionIdentifyingPayload{
+						Payload: &mfav1.SessionIdentifyingPayload_SshSessionId{SshSessionId: []byte("id")},
+					},
+					SourceCluster: "", TargetCluster: "tgt",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "missing target_cluster",
+			chal: &mfav1.ValidatedMFAChallenge{
+				Kind:     types.KindValidatedMFAChallenge,
+				Version:  "v1",
+				Metadata: &types.Metadata{Name: "test"},
+				Spec: &mfav1.ValidatedMFAChallengeSpec{
+					Payload: &mfav1.SessionIdentifyingPayload{
+						Payload: &mfav1.SessionIdentifyingPayload_SshSessionId{SshSessionId: []byte("id")},
+					},
+					SourceCluster: "src", TargetCluster: "",
+				},
+			},
+			wantErr: true,
+		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			_, err := svc.CreateValidatedMFAChallenge(t.Context(), *cmp.Or(testCase.username, &defaultUsername), testCase.chal)
+			if testCase.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
```

Comment:
Assert using ErrorContains instead of a boolean, so we know we are hitting the right conditions?

```suggestion
			if testCase.wantErr != "" {
				assert.ErrorContains(t, err, testCase.wantErr)
				return
			}
			assert.NoError(t, err)
```